### PR TITLE
compiler rework

### LINKED ISSUE #8397: 🐞 docbook_xml412 failed to install on aarch64
**Describe the bug**

docbook_xml412 failed to install on aarch64

```
docbook_xml412: A widely used XML scheme for writing documentation and help
Precompiled binary available, downloading...
Docbook_xml412 archive downloaded.
Unpacking archive using 'tar', this may take a while...
Performing pre-install...
xmlcatalog: error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory
xmlcatalog: error while loading shared libraries: libstdc++.so.6: cannot open shared object file: No such file or directory
/usr/local/lib/crew/lib/package.rb:305:in `system': Command failed with exit 127: bash
docbook_xml412 failed to install: `CFLAGS="-O2 -pipe -ffat-lto-objects -fPIC -mfloat-abi=hard -mthumb -mfpu=vfpv3-d16 -march=armv7-a+fp -fuse-ld=mold  -flto=auto" CXXFLAGS="-O2 -pipe -ffat-lto-objects -fPIC -mfloat-abi=hard -mthumb -mfpu=vfpv3-d16 -march=armv7-a+fp -fuse-ld=mold  -flto=auto" FCFLAGS="-O2 -pipe -ffat-lto-objects -fPIC -mfloat-abi=hard -mthumb -mfpu=vfpv3-d16 -march=armv7-a+fp -fuse-ld=mold  -flto=auto" FFLAGS="-O2 -pipe -ffat-lto-objects -fPIC -mfloat-abi=hard -mthumb -mfpu=vfpv3-d16 -march=armv7-a+fp -fuse-ld=mold  -flto=auto" LD_LIBRARY_PATH="/usr/local/lib" LDFLAGS="-flto=auto " bash -c xmlcatalog --noout --create /usr/local/etc/xml/docbook-xml` exited with 127
```

**To Reproduce**

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #8397 #8398
- We have an exclusion in the `crew` conflicts detection for packages named `_build` so make sure that we follow that convention.
- Fixes `crew` and `getrealdeps.sh` conflict detection so that they handle exclusions of _multiple_ `_build` filelists.
- We have `gcc_{lib,dev}` and `glibc_{lib,dev}`, so follow the `_lib` and `_dev` convention as well for llvm, renaming packages to llvm16, but retaining the old file names, as well as a `llvm` package pointing to the others.
- Update `llvm` & `openmp` to 16.0.6
- Updates `libssp` to GCC 13.1, and adds code so we don't forget about it the next time there is a GCC upgrade.
- Move most GCC libraries into `gcc_lib` from `gcc_dev` so that we don't need to keep the rest of the `gcc_dev` package around for most users. Putting GCC libraries into piecemeal packages is too much work.
- Fix `.rubocop.yml` checksum.

## Additional information

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=llvm1606 CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/gcc_dev.rb [hunk 1]
```diff
@@ -2,26 +2,27 @@
...
+  description 'The GNU Compiler Collection: Everything (excepting libraries aside from libgccjit)'
   homepage Gcc_build.homepage
-  version '13.1.0' # Do not use @_ver here, it will break the installer.
+  version '13.1.0-1' # Do not use @_ver here, it will break the installer.
   license Gcc_build.license
-  # When upgrading gcc_build, be sure to upgrade gcc_lib and gcc_dev in tandem.
-  puts "#{self} version differs from gcc version #{Gcc_build.version}".orange if version != Gcc_build.version.to_s
+  # When upgrading gcc_build, be sure to upgrade gcc_lib, gcc_dev, and libssp in tandem.
+  puts "#{self} version differs from gcc version #{Gcc_build.version}".orange if version.to_s.gsub(/-.*/,
+                                                                                                   '') != Gcc_build.version.to_s
```
Comment (uberhacker) [inline packages/gcc_dev.rb:11]: Why the line break?

File: packages/libssp.rb [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
-  version '12.2.1-b80a690'
+  version '13.1.0'
   license 'GPL-3, LGPL-3, libgcc, FDL-1.2'
+  # When upgrading gcc_build, be sure to upgrade gcc_lib, gcc_dev, and libssp in tandem.
+  puts "#{self} version differs from gcc version #{Gcc_build.version}".orange if version.to_s.gsub(/-.*/,
+                                                                                                   '') != Gcc_build.version.to_s
   compatibility 'all'
   source_url 'https://github.com/gcc-mirror/gcc.git'
-  git_hashtag 'b80a690673272919896ee5939250e50d882f2418'
+  git_hashtag 'releases/gcc-13.1.0'
```
Comment (uberhacker) [inline packages/libssp.rb:14]: `git_hashtag "releases/gcc-#{version}"`?

File: packages/llvm_build16.rb [hunk 3]
```diff
@@ -1,218 +1,15 @@
 require 'package'
+require_relative 'llvm16_build'
 
 class Llvm_build16 < Package
```
Comment (uberhacker) [inline packages/llvm_build16.rb:4]: `class Llvm16_build < Package`?

File: packages/llvm16_dev.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  depends_on 'libedit' # R
+  depends_on 'libffi' # R
+  depends_on 'libxml2' # R
+  depends_on 'llvm16_build' => :build
+  depends_on 'llvm16_lib' # R
+  depends_on 'ncurses' # R
+  depends_on 'xzutils' # R
+  depends_on 'zlibpkg' # R
+  depends_on 'zstd' # R
+  depends_on 'llvm16_build' # R
```
Comment (uberhacker) [inline packages/llvm16_dev.rb:39]: Duplicate line?

**Review by uberhacker — CHANGES_REQUESTED** [review]
Please see comments.

File: packages/gcc_dev.rb (hunk 1 above)
Comment (satmandu) [reply packages/gcc_dev.rb:11]: Linter did it...

Comment (satmandu) [reply packages/gcc_dev.rb]: Fixed.

**Review by uberhacker — APPROVED** [review]

**Pull request merged**